File: frontend/src/service/photoService.js
import { apiService, request } from "./apiService";

const PHOTO_ENDPOINT = "/photos";

const photoApiEndpoint = apiService(PHOTO_ENDPOINT);
export const photoService = {
  ...photoApiEndpoint,
  uploadPhoto: async (formData) => {
    const response = await request(PHOTO_ENDPOINT, {
      method: "POST",
      headers: {
        "Content-Type": "multipart/form-data",
      },
      data: formData,
    });
    return response;
  },

  getPhotoByEntityAndRole: async (entityId, entityType, role) => {
    const response = await request(`${PHOTO_ENDPOINT}/entity/${entityId}`, {
      method: "GET",
      params: { entityType, role },
    });
    return response;
  },

  getPhotosByEntity: async (entityId, entityType) => {
    const response = await request(`${PHOTO_ENDPOINT}/entity/${entityId}`, {
      method: "GET",
      params: { entityType },
    });
    return response;
  },

  deletePhoto: async (photoId) => {
    const response = await request(`${PHOTO_ENDPOINT}/${photoId}`, {
      method: "DELETE",
    });
    return response;
  },
};

File: backend/src/main/java/com/ibustartup/ibustartup/service/PhotoService.java
package com.ibustartup.ibustartup.service;

import com.ibustartup.ibustartup.model.Photo;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;

public interface PhotoService {
    void savePhoto(List<MultipartFile> photos, Long entityId, String entityType, String role) throws IOException;
    Long savePhotoAndReturnId(final MultipartFile photo, final Long entityId, final String entityType, final String role) throws IOException;
    void deletePhoto(Long id);
    Photo getPhotoByEntityAndRole(Long entityId, String entityType, String role);
    List<Photo> getPhotosByEntity(Long entityId, String entityType);
}
